src/memtable.rs

```diff
@@ -210,7 +215,8 @@ impl<E: Message + Clone, W: EntryExt<E>> MemTable<E, W> {
         let first = cmp::max(entries_index[0].index as usize, front);
         let last = cmp::min(entries_index[ents_len - 1].index as usize, back);
 
-        if last < front {
+        // TODO(MrCroxx): [PLEASE REVIEW] was this a bug?
+        if last < first {
```

Comment:
@hicqu PTAL, was this line a bug?

Reply:
It is, it can cause usize underflow below. You can add an unit test for it.